lib/chef/resource/windows_certificate.rb

```diff
@@ -150,12 +150,12 @@ class WindowsCertificate < Chef::Resource
           cert_obj = fetch_cert
         end
 
-        if cert_obj
+        if cert_obj != "Certificate Not Found"
```

Comment:
As above, this will always be true because cert_obj can only be true/false.

Reply:
In this case, the fetch method DOES return a Not Found

Reply:
The fetch method ultimately calls get_cert from win32-cerstore which can return a Cert Not Found

Reply:
I did update this though since it was wrong